### ISSUE TITLE
Dedicated Files Directory per Flow

### ISSUE THREAD

**Issue by MarcoGriep88** [kestra-io/kestra#4951]
### Feature description

Hi,

i just started with Kestra and found out, that the "Files" Section of Flows is shared across all flows and i unfortutaly lost some work because i did override the main.py file. I would be awsome to have separte files for each flow.

Best Regards
Marco

**Comment by anna-geller** [comment]
Hi, sorry to hear about the overridden files. [Namespace Files](https://kestra.io/docs/developer-guide/namespace-files) are files tied to a given namespace. You can think of Namespace Files as the equivalent of a project in your local IDE or a copy of your Git repository. Namespace Files are intended to hold code like Python files, R or Node.js scripts, SQL queries, dbt or Terraform projects, and many more.

To tie some files only to a specific flow, you can have a dedicated namespace for that flow e.g. `company.team.myproject` where `myproject` is what your flow does.

I understand this may not be clear at first, so let me rephrase the message shown in the UI.

**Comment by MarcoGriep88** [comment]
Hi Anna,

thanks alot. No Problem, it wasnt a big script. Thanks alot for the tip with namespace files, that helps me further :)
